contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/VoldemortBuildAndPushJob.java

```diff
@@ -212,10 +214,18 @@ public VoldemortBuildAndPushJob(String name, azkaban.utils.Props azkabanProps) {
                 if (clusterURLs.contains(url)) {
                     throw new VoldemortException("the URL: " + url + " is repeated in the "+ PUSH_CLUSTER + " property ");
                 }
-                this.clusterURLs.add(url);
-                AdminClient adminClient = createAdminClient(url, fetchAllStoresXml);
-                this.adminClientPerCluster.put(url, adminClient);
-                this.closeables.add(adminClient);
+                try {
+                    AdminClient adminClient = createAdminClient(url, fetchAllStoresXml);
+                    this.clusterURLs.add(url);
+                    this.adminClientPerCluster.put(url, adminClient);
+                    this.closeables.add(adminClient);
+                } catch (Exception e) {
```

Comment:
I've altered the `catch` logic so that we keep track of the colo(s) that failed. This will cause the job to fail with a proper error message after it is done pushing to the healthy colos.